src/qibo/quantum_info/quantum_networks.py

```
@@ -629,53 +565,497 @@ def _run_checks(self, partition, system_output, pure):
+
+    tensors = [
+        (
+            backend.to_numpy(operand.full())
+            if operand.is_pure()
+            else backend.to_numpy(operand._tensor)
+        )
+        for operand in operands
+    ]
+
+    # keep track of the `partition` and `system_input` of the network
+    _, contracrtion_list = np.einsum_path(
+        subscripts, *tensors, optimize=False, einsum_call=True
+    )
+
+    inds, idx_rm, einsum_str, remaining, blas = contracrtion_list[0]
+    input_str, results_index = einsum_str.split("->")
+    inputs = input_str.split(",")
+
+    # Warning if the same index connects two input or two output systems
+    if not surpress_warning:
+        for ind in idx_rm:
+            found = 0
+            for i, script in enumerate(inputs):
+                try:
+                    index = script.index(ind)
+                    found += 1
+                    if found > 1:
+                        if is_input != operands[inds[i]].system_input[index]:
+                            pass
+                        else:
+                            warning(
+                                f"Index {ind} connects two {'input' if is_input else 'output'} systems."
+                            )
+                    is_input = operands[inds[i]].system_input[index]
+                    if found > 2:
+                        warning(
+                            f"Index {ind} appears multiple times in the input subscripts {input_str}."
+                        )
+                except:
+                    continue
+
+    # set correct order of the `partition` and `system_input`
+    partition = []
+    system_input = []
+    for ind in results_index:
+        for i, script in enumerate(inputs):
+            try:
+                index = script.index(ind)
+
+                partition.append(operands[inds[i]].partition[index])
+                system_input.append(operands[inds[i]].system_input[index])
+
+            except:
+                continue
```

Same here

---

Ah, this is for checking the index of each letter. As the letter may not be in the script, in which case nothing needs to be done, I simply suppress the error for `script.index(ind)`